### PR TITLE
Filter project revisions API by name

### PR DESCRIPTION
**Pull request by chezou**
~This PR introduces Revision API for a specific Project. With the current implementation, there's no way to fetch the latest or specfic revision from REST API. This prevents to get user information associated with revision with the specific project.~

This PR introduces filter by name for Revisions API associated with a specific Project. With the current implementation, we don't have a way to get user info with revision without fetching revisions, but this endpoint fetches fixed number (100) of revisions and we can't fetch older revision than 100.

### CODE REVIEW

**Comment by chezou** [comment]
I know that other API endpoints are shifting to index access, so let me know if this new endpoint `api/projects/{id}/revision` isn't consistent with other APIs.

**Comment by frsyuki** [comment]
`<singular>?name=` is deprecated REST API style.
`/api/projects/{id}/revisions?name=` is consistent. It returns empty list when no matching revision is found.

**Comment by frsyuki** [comment]
What's the use case? Just checking existence of a revision?

**Comment by chezou** [comment]
The original intention was to know the latest revision's author, which is required by our customer using Python digdag client. After sending this PR I realized that `/api/projects/{id}/revisions` API returns fixed number (100) of revisions and seems to be ok to fetch the latest author from revisions  API.
Another use case could be fetching old existing revision to know the user info, but it seems nice to have thing.

**Comment by chezou** [comment]
Changed to filter revision by name.

**Comment by frsyuki** [comment]
@yoyama can I ask you to review? I think this is good

File: digdag-server/src/main/java/io/digdag/server/rs/ProjectResource.java [hunk 1]
```diff
@@ -315,13 +317,27 @@ public RestProject getProject(@PathParam("id") int projId)
...
+            List<StoredRevision> revs;
+            if (revName != null) {
+                try {
+                    StoredRevision rev = ps.getRevisionByName(proj.getId(), revName); // check NotFound first
+                    revs = ImmutableList.of(rev);
+                }
+                catch (ResourceNotFoundException ex) {
+                    revs = ImmutableList.of();
+                }
+            } else {
```
Comment (yoyama) [inline digdag-server/src/main/java/io/digdag/server/rs/ProjectResource.java]: Can we change the style as follows.
```
}
else {
```

**Comment by yoyama** [comment]
@chezou Thank you for your contribution. 
I made a comment on minor thing.
After fix it, I will approve and merge.

File: digdag-server/src/main/java/io/digdag/server/rs/ProjectResource.java (hunk 1 above)
Comment (chezou) [reply digdag-server/src/main/java/io/digdag/server/rs/ProjectResource.java]: Thanks, fixed 2aa853a

**Pull request merged**